projects/igniteui-angular/src/lib/grids/filtering/excel-style/excel-style-conditional-filter.component.ts

```diff
@@ -53,10 +53,14 @@ export class IgxExcelStyleConditionalFilterComponent implements OnDestroy {
     constructor(public esf: IgxGridExcelStyleFilteringComponent) {
         this.esf.columnChange.pipe(takeUntil(this.destroy$)).subscribe(() => {
             if (this.esf.grid) {
-                this._subMenuOverlaySettings.outlet = (this.esf.grid as any).outlet;
+                this._subMenuOverlaySettings.outlet = this.esf.grid.outlet;
             }
         });
-     }
+
+        if (this.esf.grid && !this._subMenuOverlaySettings.outlet) {
```

Comment:
`this._subMenuOverlaySettings.outlet` won't be set in the constructor.
```suggestion
        if (this.esf.grid) {
```

Reply:
It's set since the this._subMenuOverlaySettings is already instantiated, but the outlet is not.

Reply:
If `outlet` is not instantiated in the constructor why do we check it?